### ISSUE TITLE
[I] Raw ciphertext section modifyable by clicking "Insert current date"

### ISSUE THREAD

**Issue by leonggs** [pbek/QOwnNotes#2473]
#### Expected behaviour

As long as a ciphertext (within the `<!-- BEGIN ENCRYPTED TEXT --...` section) is not unlocked it should not be modifiable since otherwise it becomes useless and cannot be decrypted anymore.

#### Actual behaviour

By eg. **Insert current date** (Ctrl + T) one can accidentally insert characters into the ciphertext code. This cannot be undone by Ctrl+Z and the note cannot be encrypted anymore.
This accident can only be resolved by manual modification of the `.md` file in another text editor than QOwnNotes.

#### Steps to reproduce

Open an encrypted note in a tab. Don't decrypt it but place the cursor within the ciphertext. Execute an insert command such as **Insert current date**. The ciphertext gets modified.

#### Information about the application, settings and environment

<details><summary>Expand</summary>

QOwnNotes Debug Information
===========================

## General Info

**Current Date**: `Mo. Feb. 21 23:42:56 2022`
**Version**: `22.1.10.1`
**Build date**: `Jan 24 2022`
**Build number**: `941`
**Platform**: `linux`
**Operating System**: `Ubuntu 21.04`
**Build architecture**: `x86_64`
**Current architecture**: `x86_64`
**Release**: `Launchpad PPA`
**Qt Version (build)**: `5.15.2`
**Qt Version (runtime)**: `5.15.2`
**Portable mode**: `no`
**Application arguments**: `/usr/bin/QOwnNotes`
**Qt Debug**: `0`
**Locale (system)**: `en_US`
**Locale (interface)**: *empty*
**Primary screen resolution**: `1920x1080`
**Screen resolution(s)**: `1920x1080, 1920x1080`
**Icon theme**: `breeze-dark-qownnotes`
**Notes in current note folder**: `88`
**Calendar items**: `0`
**Enabled scripts**: `14`
</details>

**Comment by pbek** [comment]
Yes, thank you... That edit protection is currently really just an edit protection... Not a protection from any other action that inserts text.

**Comment by pbek** [comment]
That was implemented a long time ago already.
